plugin/rt_commands.go

```diff
@@ -343,3 +356,34 @@ func IsBuildDiscardArgs(args Args) bool {
 	}
 	return false
 }
+
+// Helper function to check if a command is a Maven or Gradle config command
+// These commands don't support the --insecure-tls flag
+func isMavenOrGradleConfigCommand(cmd []string) bool {
+	if len(cmd) > 0 {
+		// Check for maven and gradle config commands
+		return cmd[0] == MvnConfig || cmd[0] == GradleConfig
+	}
+	return false
+}
+
+// Helper function to determine if --insecure-tls flag should be added
+// Only add to JFrog CLI commands, not to underlying build tool commands
+func shouldAddInsecureTlsFlag(cmd []string) bool {
+	if len(cmd) == 0 {
+		return false
+	}
+	
+	// Don't add to config commands
+	if isMavenOrGradleConfigCommand(cmd) {
+		return false
+	}
+	
+	// Don't add to Maven/Gradle commands as they're passed to the underlying tool
+	if cmd[0] == MvnCmd || cmd[0] == GradleCmd {
+		return false
+	}
+	
+	// Add to all other JFrog CLI commands
+	return true
+}
```

Comment:
verify this thoroughly with the cli options as `jfrog mvn` supports it and gradle doesn't and also we shouldn't add any custom logic to make it work. 
```
➜  ~ jfrog mvn --help | grep -i insecure
  --insecure-tls        [Default: false] Set to true to skip TLS certificates verification.
➜  ~ jfrog gradle --help | grep -i insecure
```